tmt/steps/discover/fmf.py

```diff
@@ -218,12 +218,24 @@ def go(self):
         tree = tmt.Tree(path=tree_path, context=self.step.plan._fmf_context())
         self._tests = tree.tests(filters=filters, names=names)
 
+        def check_github_suffix(lib):
+            for i in range(len(lib)):
+                try:
+                    lib_url = lib[i]['url']
+                    lib_url.index('https://github.com')
+                except (TypeError, IndexError, ValueError):
+                    pass
+                else:
+                    lib[i]['url'] = lib_url.rstrip('.git')
+
         # Prefix tests and handle library requires
         for test in self._tests:
             # Prefix test path with 'tests' and possible 'path' prefix
             test.path = os.path.join(prefix_path, test.path.lstrip('/'))
             # Check for possible required beakerlib libraries
             if test.require or test.recommend:
+                check_github_suffix(test.require)
+                check_github_suffix(test.recommend)
```

Comment:
What about implementing the url suffix substitution in the `beakerlib` module which takes care of fetching libraries?
https://github.com/psss/tmt/blob/master/tmt/beakerlib.py#L74

Reply:
@psss 
if I don't miss anything the patch is obvious.
in [beakerlib.py](https://github.com/psss/tmt/blob/master/tmt/beakerlib.py#74) add 
```
            self.url = identifier.get('url')
            if self.url.find('https://github.com') != -1:
                self.url = self.url.rstrip('.git')
```
What do you think?

Reply:
Yes, this seems to be the right location. Just thinking about the `github.com` condition: Shall we enable the suffix stripping for all repositories? Just tried with pagure, github and gitlab and all of them strip the suffix when provided in the url. @lukaszachy, @hegerj, @thrix, what do you think?

Reply:
@dkarpele I think `str.startswith` is faster than `str.find` and it returns bool directly.

@psss I'd be careful with 'all', [git-scm book](https://git-scm.com/book/en/v2/Git-on-the-Server-Getting-Git-on-a-Server) says that '.git' suffix is a convention for bare repositories. I guess stripping pagure/gitlab/github shouldn't hurt  - Once plans can be inherited the likelihood of mismatch between libraries will increase

Reply:
Ok, could we agreed to strip only github.com? We can remove this condition later if it's necessary.

Reply:
> I guess stripping pagure/gitlab/github shouldn't hurt

Agreed. I'd suggest to go with these three for now.